src/core/Akka/Dispatch/CurrentSynchronizationContextDispatcher.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+
+namespace Akka.Dispatch
+{
+    /// <summary>
+    /// INTERNAL API
+    /// 
+    /// Produces <see cref="ExecutorService"/> that dispatches messages on the current synchronization context,
+    ///  e.g. WinForms or WPF GUI thread
+    /// </summary>
+    internal sealed class CurrentSynchronizationContextExecutorServiceFactory : ExecutorServiceConfigurator
+    {
+        public override ExecutorService Produce(string id)
+        {
+            return new TaskSchedulerExecutor(id, TaskScheduler.FromCurrentSynchronizationContext());
+        }
+
+        public CurrentSynchronizationContextExecutorServiceFactory(Config config, IDispatcherPrerequisites prerequisites) : base(config, prerequisites)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Used to create instances of the <see cref="PinnedDispatcher"/>. 
+    /// 
+    /// Each actor created using the pinned dispatcher gets its own unique thread.
+    /// <remarks>
+    /// Always returns a new instance.
+    /// </remarks>
+    /// </summary>
+    internal sealed class CurrentSynchronizationContextDispatcherConfigurator : MessageDispatcherConfigurator
```

Comment:
Decided to wrap the `CurrentSynchronizationContextDispatcher` into its own class - it needs similar semantics to the `PinnedDispatcher` (this dispatcher belongs to exactly 1 actor instance) and it also needs to be used exclusively with a single executor type.